api_test.go

```diff
@@ -386,3 +394,135 @@ func TestAwsServiceFactoryGetLightsailService(t *testing.T) {
 		}
 	}
 }
+
+func TestAwsServiceFactoryGetEKSService(t *testing.T) {
+	// Create our test cases
+	cases := []struct {
+		RegionName string
+	}{
+		{},
+		{
+			RegionName: "us-west-1",
+		},
+	}
+
+	// Loop through the test cases
+	for _, c := range cases {
+		// Create a config for the region?
+		var config = &aws.Config{}
+		if c.RegionName != "" {
+			config = config.WithRegion(c.RegionName)
+		}
+
+		// Create our test
+		session, err := session.NewSession(config)
+		if err != nil {
+			t.Errorf("Unexpected error while creating a new session: %v", err)
+		}
+
+		// Create an AWS Service Factory
+		sf := &AWSServiceFactory{
+			Session: session,
+		}
+
+		// Get the desired service
+		service := sf.GetEKSService(c.RegionName)
+
+		// Is the service nil?
+		if service == nil {
+			t.Errorf("No service returned for %s", "GetLightsailService")
+		} else if service.Client != nil {
+			// Convert to implementation type
+			implType, ok := service.Client.(*eks.EKS)
+			if !ok {
+				t.Errorf("Unexpected Client type: expected %v, actual %v", "*eks.EKS", implType)
+			} else if *implType.Config.Region != c.RegionName {
+				t.Errorf("Unexpected value for Client.Config.Region: expected %s, actual %s", c.RegionName, *implType.Config.Region)
+			}
+		}
```

Comment:
Suggestion _(non-blocking)_:
By wrapping the eksservice creation & assertions on the result inside of a `t.Run` func, the output of a failing test will point you to the specific test case that failed, rather than just the test func as a whole (TestAwsServiceFactoryGetEKSService)
```suggestion
        t.Run(fmt.Sprintf("testing with region name: %s", c.RegionName), func(t *testing.T) {
		    // Get the desired service
		    service := sf.GetEKSService(c.RegionName)
    
		    // Is the service nil?
		    if service == nil {
			    t.Errorf("No service returned for %s", "GetLightsailService")
		    } else if service.Client != nil {
			    // Convert to implementation type
			    implType, ok := service.Client.(*eks.EKS)
			    if !ok {
				    t.Errorf("Unexpected Client type: expected %v, actual %v", "*eks.EKS", implType)
			    } else if *implType.Config.Region != c.RegionName {
				    t.Errorf("Unexpected value for Client.Config.Region: expected %s, actual %s", c.RegionName, *implType.Config.Region)
			    }
		    }
		  })
```

Reply:
& for more context, you can put the entire content of the for loop inside of this `t.Run` block to achieve the same results, but I personally like to limit to the section where the tested function is executed & the assertions are made. But that's just preferential.